xcdat/bounds.py

```diff
@@ -603,7 +603,10 @@ def _create_time_bounds(
         elif freq == "hour":
             # Determine the daily frequency for generating time  bounds.
             if daily_subfreq is None:
-                diff = time.values[1] - time.values[0]
+                diff = pd.to_timedelta(time.values[1] - time.values[0])
+                # `cftime` objects only support arithmetic using `timedelta` objects, so
+                # the values of  `diffs` must be casted from `dtype="timedelta64[ns]"`
+                # to `timedelta` objects.
```

Comment:
Suggest putting the comment ahead of the code (as is done earlier in the [code](https://github.com/xCDAT/xcdat/blob/223869e2a286ae403427e26921c749f60b0d4f6b/xcdat/bounds.py#L916)).

Reply:
I was initially averse to adding in a pandas converter here, because in a different project I've run into issues with this (the behavior of the datetime conversion is sensitive to the input array), but your pointing out that this is already done earlier in the code makes me way less worried about this. 

In the [code block](https://github.com/xCDAT/xcdat/blob/223869e2a286ae403427e26921c749f60b0d4f6b/xcdat/bounds.py#L912) you pointed out in the issue, this conversion is conditioned on the array being a time axis (not needed here) and the array containing `cftime` objects. It does seem like it might prevent problems to only do this conversion if necessary (i.e., add a conditional here).

But I am a little confused because the existing conditional is meant to apply to `cftime`-based axes (and here it is meant to apply to axes that do not have `cftime` objects). 

I'm curious about:
* should there be a conditional to do this type conversion?
* should the conditional be applied if the object `type(time.values[0]) == np.datetime64)` or `not contains_cftime_datetimes(xr.as_variable(time))`?

Reply:
@pochedls thanks for the suggestion. I initially have coded as like below:
```python
if isinstance(diff, np.timedelta64):
    diff = pd.to_timedelta(diff)
```
but this interrupted the GitHub Action CI check process for build, which was complaining that the function is too complicated. It seem like the CI process didn't like having more than a certain number of `if` layers. 

I tested the code with a CMIP model data that has cftime-based time axis, and the same operation was running okay without errors, so was presuming it was okay with cftime-based axes. But I am open to any suggestion to make the code more robust.

Reply:
I like this... @tomvothecoder may be able to suggest some a way to pass the CI tests (or skip them for that line, e.g., [here](https://github.com/xCDAT/xcdat/blob/main/xcdat/bounds.py#L579)).

Reply:
I also like it but it does pass [flake8 C901](https://www.flake8rules.com/rules/C901.html) with the error message of `xcdat/bounds.py:507:5: C901 'BoundsAccessor._create_time_bounds' is too complex (11)`

`# type: ignore` or `#noqa: C901` (following the [Stackoverflow](https://stackoverflow.com/questions/51127170/flake8-ignore-warnings-for-a-function) didn't help.

Reply:
I updated the code to have the if statement but as expected it fails flake8 checks. @tomvothecoder Any help will be appreciated.

Reply:
I added `#noqa: C901` and updated the comment inside `_create_time_bounds()` here: https://github.com/xCDAT/xcdat/pull/660/commits/4c290e7435cf6a5832fd5f480548699612a7c5ca.

I can help with adding the unit test tomorrow, which you can hopefully try out in #655 too.